src/theory/smt_engine_subsolver.cpp

```diff
@@ -41,60 +41,40 @@ Result quickCheck(Node& query)
   return Result(Result::SAT_UNKNOWN, Result::REQUIRES_FULL_CHECK);
 }
 
-void initializeSubsolver(SmtEngine* smte)
+void initializeSubsolver(std::unique_ptr<SmtEngine>& smte,
```

Comment:
I have a slight preference for actually just returning a `std::unique_ptr<SmtEngine>` but we can address that later.

Reply:
Agreed